src/support/archive.cpp

```diff
@@ -129,11 +129,8 @@ std::string Archive::Child::getRawName() const {
 }
 
 Archive::Child Archive::Child::getNext(bool& error) const {
-  size_t toSkip = len;
-  // Members are aligned to even byte boundaries.
-  if (toSkip & 1) ++toSkip;
-  const uint8_t* nextLoc = data + toSkip;
-  if (nextLoc >= (uint8_t*)&*parent->data.end()) {  // End of the archive.
+  const uint8_t* nextLoc = data + len + (len & 1); // Members are aligned to even byte boundaries.
+  if (nextLoc >= (const uint8_t*)parent->data.data() + parent->data.size()) {  // End of the archive.
```

Comment:
Technically even having a pointer that points outside an allocation is UB (see http://stackoverflow.com/questions/10473573/why-is-out-of-bounds-pointer-arithmetic-undefined-behaviour) which is why i did the cryptic version. Practically, it's just supposed to point at 1-past-the end, which should be ok, but... you know, paranoia.